go.mod

```diff
@@ -7,6 +7,7 @@ require (
 	github.com/gobuffalo/flect v0.1.5
 	github.com/golang/protobuf v1.3.1 // indirect
 	github.com/kr/pretty v0.1.0 // indirect
+	github.com/mitchellh/mapstructure v1.2.2
```

Comment:
I understand that Kubebuilder has a straight politic to add new dependencies for some reasons as for example keep the Maintenance and supportability ability. 

Looking at the dep https://github.com/mitchellh/mapstructure I am not sure if it is something really adopted by the community and that could/should be added here(IMO we should NOT use the mitchellh's module). So, if it has no other way to do be solved whiteout this module then, I believe that this PR will require the admin approvals. @mengqiy wdyt?

Reply:
We copied validations because it was trivial.  Non-trivial dependencies are fine.  That being said: this code already more-or-less exists in kubernetes in the unstructured codec, if we want to use it (unclear that it makes much of a difference).